src/Microsoft.AspNetCore.Dispatcher/ITemplateEndpoint.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Dispatcher
+{
+    public interface ITemplateEndpoint
+    {
+        string HttpMethod { get; }
+
+        string Template { get; }
+
+        DispatcherValueCollection Values { get; }
```

Comment:
Still not sure why this is available on the end point

Reply:
These are the values that need to be present to select the endpoint. We can go further here by just doing token replacement

Reply:
Even if we do token replacement this property still has to be here. Attribute routing treats these like defaults, that's required for ambient values to work